azure-quantum/azure/quantum/qiskit/backends/rigetti.py

```diff
@@ -0,0 +1,117 @@
+##
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+##
+import warnings
+
+from typing import TYPE_CHECKING, Union, List
+from azure.quantum.version import __version__
+from azure.quantum.qiskit.job import AzureQuantumJob
+from azure.quantum.target.rigetti import RigettiTarget
+
+from .backend import AzureBackend
+
+from qiskit.providers.models import BackendConfiguration
+from qiskit.providers import Options
+from qiskit_qir import SUPPORTED_INSTRUCTIONS as QIR_BASIS_GATES
+
+if TYPE_CHECKING:
+    from azure.quantum.qiskit import AzureQuantumProvider
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "RigettiSimulatorBackend"
+    "RigettiQPUBackend"
+]
+
+class RigettiBackend(AzureBackend):
+    """Base class for interfacing with a Rigetti backend in Azure Quantum"""
+
+    @classmethod
+    def _default_options(cls):
+        return Options(count=500, entryPoint="main", arguments=[], targetCapability="BaseProfileExecution")
+
+    @classmethod
+    def _azure_config(cls):
+        return {
+            "blob_name": "inputData",
+            "content_type": "qir.v1",
+            "provider_id": "rigetti",
+            "input_data_format": "qir.v1",
+            "output_data_format": "microsoft.quantum-results.v1",
+        }
+
+
+class RigettiSimulatorBackend(RigettiBackend):
+    backend_names = (
+        RigettiTarget.QVM.value,
+    )
+
+    def __init__(
+        self,
+        name: str,
+        provider: "AzureQuantumProvider",
+        **kwargs
+    ):
+        """Base class for interfacing with an Rigetti Simulator backend"""
+        default_config = BackendConfiguration.from_dict(
+            {
+                "backend_name": name,
+                "backend_version": __version__,
+                "simulator": True,
+                "local": False,
+                "coupling_map": None,
+                "description": "Rigetti simulator on Azure Quantum",
+                "basis_gates": QIR_BASIS_GATES,
+                "memory": False,
+                "n_qubits": 38,
```

Comment:
Is this a limit on the number of qubits that can be simulated? If so, I think we have to lower it based on QVM's usage. I can get you an exact number with a bit of digging but I think it will be closer to 20.

Reply:
ok. lowered it to 20.